src/app/clusters/commodity-tariff-server/commodity-tariff-server.h

```diff
@@ -0,0 +1,172 @@
+
+    void SetEndpointId(EndpointId aEndpoint) { mEndpointId = aEndpoint; }
+
+
+    virtual Protocols::InteractionModel::Status GetDayEntryById(DataModel::Nullable<uint32_t> aDayEntryId,
+                                                                Structs::DayEntryStruct::Type & aDayEntry) = 0;
+
+    virtual Protocols::InteractionModel::Status GetTariffComponentInfoById(DataModel::Nullable<uint32_t>  aTariffComponentId,
+                                                                           DataModel::Nullable<chip::CharSpan> & label,
+                                                                           DataModel::List<const uint32_t> & dayEntryIDs,
+                                                                           Structs::TariffComponentStruct::Type & aTariffComponent) = 0;
+
+    // ------------------------------------------------------------------
+#if 0
+    // Attribute  getters
+    virtual DataModel::Nullable<Structs::TariffInformationStruct::Type> & GetTariffInfo()        = 0;
+    virtual DataModel::Nullable<Globals::TariffUnitEnum> & GetTariffUnit()                       = 0;
+    virtual DataModel::Nullable<uint32_t> & GetStartDate()                                       = 0;
+    virtual DataModel::Nullable<int16_t> & GetDefaultRandomizationOffset()                       = 0;
+    virtual DataModel::Nullable<DayEntryRandomizationTypeEnum> & GetDefaultRandomizationType()   = 0;
+    virtual DataModel::List<Structs::CalendarPeriodStruct::Type> & GetCalendarPeriods()          = 0;
+    virtual DataModel::List<Structs::DayPatternStruct::Type> & GetDayPatterns()                  = 0;
+    virtual DataModel::List<Structs::DayStruct::Type> & GetIndividualDays()                      = 0;
+    virtual DataModel::List<Structs::DayEntryStruct::Type> & GetDayEntries()                     = 0;
+    virtual DataModel::List<Structs::TariffPeriodStruct::Type> & GetTariffPeriods()              = 0;
+    virtual DataModel::List<Structs::TariffComponentStruct::Type> & GetTariffComponents()        = 0;
+    virtual DataModel::Nullable<Structs::DayStruct::Type> & GetCurrentDay()                      = 0;
+    virtual DataModel::Nullable<Structs::DayStruct::Type> & GetNextDay()                         = 0;
+    virtual DataModel::Nullable<Structs::DayEntryStruct::Type> & GetCurrentDayEntry()            = 0;
+    virtual DataModel::Nullable<Structs::DayEntryStruct::Type> & GetNextDayEntry()               = 0;
+    virtual DataModel::Nullable<uint32_t> & GetCurrentDayEntryDate()                             = 0;
+    virtual DataModel::Nullable<uint32_t> & GetNextDayEntryDate()                                = 0;
+    virtual DataModel::List<Structs::TariffComponentStruct::Type> & GetCurrentTariffComponents() = 0;
+    virtual DataModel::List<Structs::TariffComponentStruct::Type> & GetNextTariffComponents()    = 0;
+#else
+    #define X(attrName, attrType) \
+        virtual attrType& Get##attrName() = 0;
```

Comment:
The simple data bits should probably live in the cluster, not be gotten from the app every time, because that's a recipe for subscriptions being broken.  If the app updates the cluster impl when those attributes change, the cluster impl can handle the firty marking and whatnot.

For lists the situation is more complicated, of course, unless the cluster impl copies the list.